%%
%% Copyright 2007, 2008, 2009 Elsevier Ltd
%%
%% This file is part of the 'Elsarticle Bundle'.
%% ---------------------------------------------
%%
%% It may be distributed under the conditions of the LaTeX Project Public
%% License, either version 1.2 of this license or (at your option) any
%% later version.  The latest version of this license is in
%%    http://www.latex-project.org/lppl.txt
%% and version 1.2 or later is part of all distributions of LaTeX
%% version 1999/12/01 or later.
%%
%% The list of all files belonging to the 'Elsarticle Bundle' is
%% given in the file `manifest.txt'.
%%

%% Template article for Elsevier's document class `elsarticle'
%% with numbered style bibliographic references
%% SP 2008/03/01
%%
%%
%%
%% $Id: elsarticle-template-num.tex 4 2009-10-24 08:22:58Z rishi $
%%
%%

%\documentclass[preprint,12pt]{elsarticle}

%% Use the option review to obtain double line spacing
%% \documentclass[preprint,review,12pt]{elsarticle}

%% Use the options 1p,twocolumn; 3p; 3p,twocolumn; 5p; or 5p,twocolumn
%% for a journal layout:
%% \documentclass[final,1p,times]{elsarticle}
%% \documentclass[final,1p,times,twocolumn]{elsarticle}
%% \documentclass[final,3p,times]{elsarticle}
%% \documentclass[final,3p,times,twocolumn]{elsarticle}
%% \documentclass[final,5p,times]{elsarticle}
\documentclass[final,5p,times,twocolumn]{elsarticle}

%% if you use PostScript figures in your article
%% use the graphics package for simple commands
%% \usepackage{graphics}
%% or use the graphicx package for more complicated commands
\usepackage{graphicx}
%% or use the epsfig package if you prefer to use the old commands
%% \usepackage{epsfig}

%% The amssymb package provides various useful mathematical symbols
\usepackage{amssymb}
%% The amsthm package provides extended theorem environments
%% \usepackage{amsthm}

%% The lineno packages adds line numbers. Start line numbering with
%% \begin{linenumbers}, end it with \end{linenumbers}. Or switch it on
%% for the whole article with \linenumbers after \end{frontmatter}.
%% \usepackage{lineno}

%% natbib.sty is loaded by default. However, natbib options can be
%% provided with \biboptions{...} command. Following options are
%% valid:

%%   round  -  round parentheses are used (default)
%%   square -  square brackets are used   [option]
%%   curly  -  curly braces are used      {option}
%%   angle  -  angle brackets are used    <option>
%%   semicolon  -  multiple citations separated by semi-colon
%%   colon  - same as semicolon, an earlier confusion
%%   comma  -  separated by comma
%%   numbers-  selects numerical citations
%%   super  -  numerical citations as superscripts
%%   sort   -  sorts multiple citations according to order in ref. list
%%   sort&compress   -  like sort, but also compresses numerical citations
%%   compress - compresses without sorting
%%
%% \biboptions{comma,round}

% \biboptions{}


\journal{Journal of Chromatography B}

\begin{document}

\begin{frontmatter}

%% Title, authors and addresses

%% use the tnoteref command within \title for footnotes;
%% use the tnotetext command for the associated footnote;
%% use the fnref command within \author or \address for footnotes;
%% use the fntext command for the associated footnote;
%% use the corref command within \author for corresponding author footnotes;
%% use the cortext command for the associated footnote;
%% use the ead command for the email address,
%% and the form \ead[url] for the home page:
%%
%% \title{Title\tnoteref{label1}}
%% \tnotetext[label1]{}
%% \author{Name\corref{cor1}\fnref{label2}}
%% \ead{email address}
%% \ead[url]{home page}
%% \fntext[label2]{}
%% \cortext[cor1]{}
%% \address{Address\fnref{label3}}
%% \fntext[label3]{}

%& SHORT COMMUNICATION about 2850 words 
\title{Straightforward interpretation of metabolomics, proteomics, transcriptomics, and genomics data by comprehensive visualization and pathway enrichment using a single software tool}

%% use optional labels to link authors explicitly to addresses:
%% \author[label1,label2]{<author name>}
%% \address[label1]{<address>}
%% \address[label2]{<address>}

\author[uni]{Lars Rosenbaum\corref{cor}}
\ead{lars.rosenbaum@uni-tuebingen.de}
\author[uni]{Johannes Eichner}
\author[uni]{Clemens Wrzodek}
\author[endo,dzd]{Hans-Ulrich H\"aring}
\author[uni]{Andreas Zell}
\author[zentrallabor,dzd]{Rainer Lehmann\corref{cor}}
\ead{rainer.lehmann@med.uni-tuebingen.de}
\address[uni]{Center for Bioinformatics, University of T\"ubingen, T\"ubingen, Germany}
\address[dzd]{Institute of Diabetes Research and Metabolic Diseases, Member of the German Center for Diabetes Research, University of T\"ubingen, T\"ubingen, Germany}
\address[endo]{Department of Internal Medicine, Division of Endocrinology, Diabetology, Vascular Medicine, Nephrology and Clinical Chemistry, University Hospital T\"ubingen, T\"ubingen, Germany}
\address[zentrallabor]{Division of Clinical Chemistry and Pathobiochemistry (Central Laboratory), University Hospital T\"ubingen, T\"ubingen, Germany}


\cortext[cor]{Corresponding authors. Tel. +49-7071-29-8 31 93}

\begin{abstract}
%% Text of abstract
In systems biology, the combination of multiple types of omics data, such as metabolomics, proteomics, transcriptomics, and genomics, yields more information on a biological process than the analysis of a single type of data. Thus, data from different omics platforms is usually combined in one experimental setup to obtain insight into a biological process or a disease state. Particularly high accuracy metabolomics data from modern mass spectrometry instruments is currently more and more integrated into biological studies. Reflecting this trend, we extended InCroMAP to realize the integration of metabolomics data. Now, the tool is able to perform an integrated enrichment analysis and pathway-based visualization of multi-omics data and thus suitable for the evaluation of comprehensive systems biology studies.
\end{abstract}

\begin{keyword}
%% keywords here, in the form: keyword \sep keyword
Metabolomics \sep Transcriptomics \sep Proteomics \sep Systems biology \sep Enrichment analysis \sep Pathway visualization \sep Bioinformatics
%% MSC codes here, in the form: \MSC code \sep code
%% or \MSC[2008] code \sep code (2000 is the default)

\end{keyword}

\end{frontmatter}

%%
%% Start line numbering here if you want
%%
% \linenumbers

%% main text
\section{Introduction}
Today, high-throughput methods for the analysis of biological systems, such as microarrays, next generation sequencing, and mass spectrometry generate a wealth of omics-scale data. To develop hypotheses about a biological process or a disease state, a variety of omics platforms for measuring different genomic, proteomic, and metabolic features are combined in one experimental setup. Probably the most prominent genomic feature is messenger RNA (mRNA), which can be quantified on a genome-wide level by gene expression chips (microarrays) or next generation sequencing techniques. Other important (epi)genomic features include microRNA (miRNA), single-nucleotide polymorphisms, and epigenetic information, such as the promoter methylation status (DNAm). Proteomic features can be derived from abundance profiling of particular protein isoforms and posttranslational modifications, such as methylation and phosphorylation. Furthermore, modern mass spectrometry platforms are able to detect and measure the relative intensity of thousands of metabolites with high accuracy. The sensible and integrated visualization of omics data at different levels of abstraction is crucial to obtain biological insight without being overwhelmed by the intrinsic complexity of the data \cite{Gehlenborg2010}. A key concept for detecting alterations in cell signalling or metabolism in a biological system are pathway-based visualizations.

A plethora of tools were developed for the inspection of data from individual platforms (see \cite{Gehlenborg2010} for examples). Furthermore, solutions for the combined visualization of transcriptomics (mRNA) and metabolomics data exist \cite{Garcia-Alcalde2011,Waegele2012}. Both Paintomics \cite{Garcia-Alcalde2011} and MassTRIX \cite{Waegele2012} are web-service based tools that are capable of visualizing mRNA microarray and identified metabolomics data in KEGG pathways \cite{Kanehisa2006}. MassTRIX is able to handle unidentified metabolic data by comparing each mass against theoretical adducts stored in metabolomics databases. An example of a tool that can handle several heterogeneous types of omics data is the commercial Ingenuity Pathway Analysis software (www.ingenuity.com). However, Ingenuity does not provide an integrated visualization of multi-level omics data. In summary, the comprehensive analysis of multi-omics data is presently limited because current high-level analysis tools are not able to perform an integrated analysis of data from multiple types of omics platformsare, focused on certain specialized platforms, or not freely available. Thus, novel analysis tools and appropriate visualizations are required, since complex interactions between multiple layers of a biological system can only be inferred by the integration of omics data sets across multiple platforms.

In this contribution we present an extended version of the tool InCroMAP \cite{Wrzodek2012a,Wrzodek2012b}. InCroMAP is a stand-alone Java software originally developed for the enrichment analysis and pathway-based visualizations of genomic and proteomic data, where multiple biological layers were monitored in the same set of samples. The application was designed to provide a high ease of use. Consequently, all information required, for example, for the mapping between different metabolite identifiers or the annotation of miRNAs with mRNA targets, is either directly included in the tool or dynamically downloaded in the background. Previous versions of InCroMAP focused on genomics and transcriptomics data were designed for the combined analysis of a wide variety of omics platform types, where each feature can be linked to a gene or genomic interval. Up to the present, metabolomics data was not supported. In the extended version of InCroMAP presented here, we support the enrichment analysis and pathway-based visualization of annotated metabolomics data and thereby complemented the tool for comprehensive systems biology data evaluation. For convenience, InCroMAP supports several commonly used compound identifiers, such as specific database identifiers (e.g., HMDB \cite{Wishart2009}), common synonyms, and IUPAC names. 

InCroMAP is freely available under the LGPL3 license at http://www.cogsys.cs.uni-tuebingen.de/software/InCroMAP, including a comprehensive userâ€™s guide and several example data files to test the capabilities of the tool.

\section{Methods}
Before systems biology data from heterogeneous platforms can be integrated and visualized with high-level data analysis tools like InCroMAP, the data has to be preprocessed in a platform-dependent manner (see Figure \ref{fig:incromap-workflow}). A typical workflow for metabolomics data from nontargeted mass spectrometry measurements includes the following steps. First, metabolic features, which are characterized by mass and relative intensity, are extracted from the raw data files with feature finding and feature extraction algorithms. Ideally, the feature induced by a metabolite should contain all signals that were generated by the metabolite (including isotopic peaks). Second, the extracted features have to be aligned and linked between the different samples. The aforementioned preprocessing steps can be performed with sophisticated open-source software tools like OpenMS \cite{Sturm2008}. The  features are then subjected to quality controls and low-level statistical data analysis, which includes normalization and calculation of measures for differential abundance between conditions (e.g., p-values, fold changes, or log ratios). Mostly, these tasks are performed with commercial statistics software, open-source omics analysis applications like Mayday \cite{Battke2010}, or directly with the statistical programming language R (www.r-project.org). In a final preprocessing step, the metabolite features are annotated with candidate identifiers using information provided by metabolomics databases, such as PubChem \cite{Wang2009}, HMDB \cite{Wishart2009}, or LIPID MAPS \cite{Sud2007}. Similar preprocessing workflows exist for other platforms, like microarrays, next generation sequencing, and proteomics. The processed data can then be imported and analyzed with InCroMAP.

In a typical use case of InCroMAP (see Figure \ref{fig:incromap-workflow}), the user first imports his preprocessed multi-level omics data, given in tabular format. Then, the differentially expressed metabolites or genes are determined for each platform based on appropriate cutoffs. Next, relevant pathways related to the biological background of the experiments are inferred. For the detection of relevant pathways, InCroMAP employs a special pathway enrichment algorithm which integrates differentially expressed metabolites, genes, and proteins across multiple platforms. The resulting pathways can then be selected for further visual inspection from a table in which each pathway is associated with a significance value. Alternatively, the metabolic overview function of InCroMAP can be used to generate an interactive global map of cellular metabolism, in which each subordinate metabolic pathway is colored according to the significance of its enrichment. The results of the enrichment analysis can be exported in tabular format and the pathway-based visualizations can be easily stored as JPEG images.

\begin{figure*}
\center
\includegraphics[width=0.85\textwidth]{InCroMAP_workflow.pdf}
\caption{The flow chart depicts the general workflow of an analysis with InCroMAP. The raw data from each platform is preprocessed using appropriate, platform-specific algorithms for data preparation (e.g., feature extraction) and the alleviation of experimental artifacts (e.g., normalization techniques). Then, the data is analyzed with a statistics software, which yields p-values or fold changes for each of the features. Finally, the gene, protein, and metabolite features are annotated with candidate identifiers. After these preprocessing steps the data can be easily imported to InCroMAP and an enrichment analysis can be performed. On the basis of the enrichment, the user may select a pathway of interest, which can be overlaid with the corresponding omics data. Alternatively, a visualization showing a global overview of the alterations in metabolic pathways can be generated.}
\label{fig:incromap-workflow}
\end{figure*}

\begin{figure*}[t]
\center
\includegraphics[width=1.0\textwidth]{InCroMAP_examples.pdf}
\caption{(A) shows the metabolic overview with visualized metabolic data and pathway enrichment p-values. (B) depicts the Wnt signalling pathway with visualized data from all supported platforms. Both illustrations were generated with the example data supplied on the project web page. (c) shows a legend that explains the visualization of the data types and the coloring schemes.}
\label{fig:incromap-examples}
\end{figure*}

\subsection{Data import}
For an import to InCroMAP the data has to be in a tabular format obtained, e.g., by MS Excel export functions. Furthermore, each measurement must contain at least some identifier (e.g., HMDB ID) and an observation (e.g., p-value, fold-change, or log ratio) that captures the difference between experimental conditions. InCroMAP is able to handle various types of identifiers for transcriptomics, proteomics, and metabolomics data. It supports the automatic recognition of identifiers used by the most common oligonucleotide microarray manufacturers (e.g., Affymetrix, Agilent, etc.) and generic formats to enable the import of processed data, provided that each measurement can be either associated to a certain gene, genomic region, protein, or metabolite. Protein data is mapped to a gene for the enrichment analysis and visualization. Thus, proteomics data has to be labeled with a gene identifier and additionally with a name that states the protein modification. Metabolomics data may be annotated by a database identifier (e.g., KEGG, HMDB, PubChem, LIPID MAPS), a common synonym, or a IUPAC name. Pathways of interest can either be automatically downloaded from KEGG or imported from other sources in BioPAX format.

\subsection{Enrichment analysis}
A common algorithm for performing an enrichment of gene, protein, or metabolite sets is an overrepresentation analysis \cite{Backes2007}, which uses a hypergeometric test to calculate a significance value for each predefined set. InCroMAP extends the method to provide an integrated enrichment across multiple platforms. The overrepresentation analysis requires a fixed set of up- or downregulated genes, proteins, or metabolites, which are obtained by a user-defined fold change and/or p-value cutoff for each of the platform-dependent data sets. In the presented version, InCroMAP supports the enrichment of KEGG pathways (including proteins/compounds), of gene sets from the molecular signatures database (MSigDB), and of gene ontology (GO) terms. The result list of an exemplary KEGG pathway enrichment is depicted in Figure \ref{fig:incromap-workflow}, where each pathway is assigned a p-value and an FDR corrected p-value (q-value).

During the enrichment analysis, InCroMAP discards unidientified features that could not be mapped to a gene or metabolite. DISCUSS WITH RAINER

\subsection{Metabolic overview}
The results of a pathway enrichment analysis are typically presented to the user as a table or barplot of pathways, sorted according to the associated p-values, which score the significance of the enrichment with deregulated genes. As these traditional representations of an enrichment result do not highlight any functional relations between pathways we devised an alternative method, which provides the user with a more structured view of the metabolic pathways altered by a certain condition. Using the metabolic overview feature, the user can display enrichment results in the context of the KEGG's global `Metabolic Pathways' map (KEGG: map01100), which contains references to all subordinate metabolic KEGG pathways. In this illustration pathways which are significantly enriched with up- or downregulated genes or metabolites are highlighted in different shades of blue (Figure 2A). Since KEGG does not offer a meta-pathway for cellular signal transduction, this feature is limited to metabolic pathways.

\subsection{Pathway-based visualization}
Metabolic or signalling pathways of interest can either be automatically downloaded from KEGG or imported from other sources in BioPAX format. Being rendered in an interactive graph viewer, the pathway nodes representing genes/proteins can be overlaid with expression data from mRNAs and multiple protein products (Figure 2B). Additionally, miRNAs can be connected to a given pathway based on experimentally confirmed or predicted interactions to their target mRNAs. If desired, the tool also visualizes differential methylation of proximal gene promoters, which is by default computed based on the largest peak observed in the upstream region of the transcription start site. In the extended version of InCroMAP we also support the visualization of metabolomics data. For each pathway node corresponding to a gene, protein, or metabolite, the individual measurements derived from multiple platforms and/or probesets as well as the DNA methylation profile of the proximal promoter can be inspected in a separate details panel. 

\section{Results and discussion}
We presented an extended version of InCroMAP, which allows the integrated pathway-based visualization of data from multiple omics platforms, such as DNAm, mRNA, miRNA, proteins, protein modifications, and metabolome data. The tool enables a user to interactively browse through and visualize different KEGG pathways. Furthermore, the metabolic overview function of InCroMAP can be used to generate an interactive global map of cellular metabolism, which provides a structured view of the metabolic changes present in a certain experimental condition. The tool provides a useful overview of multi-omics data in the context of metabolic and signalling pathways, which allows a user to gain insight in complex multi-omics data sets. Thus, it facilitates the interpretation of otherwise cumbersome data and enables the generation of initial biological hypotheses. 

With the ability to handle nontargeted metabolimics data, results in limitations of the current pathway enrichment method. The annotation of nontargeted metabolomics data sets usually yields many-to-many mappings because a metabolic feature can be mapped to several candidate identifiers and vice versa. The overrepresentation analysis is not designed to handle such many-to-many mappings. Solutions to this problem in the setting of gene set enrichment have been proposed \cite{Kankainen2011} and will be implemented in a future version of InCroMAP. 

Further future improvements concern the integration of visualizations from additional metabolomics and lipidomics databases, such as Reactome \cite{Eustachio2011} and LIPID MAPS \cite{Sud2007}. Particularly the inclusion of visualizations based on LIPID MAPS pathways can overcome limitations of KEGG with respect to lipidomics data. Additionally, an automatic annotation with candidate identifiers for nontargeted metabolomics data based on the comparison against theoretical adducts of metabolites from metabolomic databases is planned for a future release of InCroMAP. The tool automatically notifies a user if an update is available on the project homepage (http://www.cogsys.cs.uni-tuebingen.de/software/InCroMAP).

To conclude, we think that InCroMAP is a useful tool for the straightforward analysis and visualization of complex metabolomics, proteomics, transcriptomics, and genomics data.

\section{Acknowledgements}
This project was supported by the Competence Network for Diabetes mellitus funded
by the BMBF (FkZ 01 GI 1104A) and a grant from to the German Center for Diabetes Research (DZD eV). Furthermore, the project received funding from the Innovative Medicine Initiative Joint Undertaking (IMI JU) [115001] (MARCAR project).

%% The Appendices part is started with the command \appendix;
%% appendix sections are then done as normal sections
%% \appendix

%% \section{}
%% \label{}

%% References
%%
%% Following citation commands can be used in the body text:
%% Usage of \cite is as follows:
%%   \cite{key}         ==>>  [#]
%%   \cite[chap. 2]{key} ==>> [#, chap. 2]
%%

%% References with bibTeX database:


\bibliographystyle{elsarticle-num}
\bibliography{literature}

%% Authors are advised to submit their bibtex database files. They are
%% requested to list a bibtex style file in the manuscript if they do
%% not want to use elsarticle-num.bst.

%% References without bibTeX database:

% \begin{thebibliography}{00}

%% \bibitem must have the following form:
%%   \bibitem{key}...
%%

% \bibitem{}

% \end{thebibliography}


\end{document}

%%
%% End of file `elsarticle-template-num.tex'.
